contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -3416,9 +3416,44 @@ LANGUAGE C STABLE PARALLEL SAFE;
 CREATE OR REPLACE FUNCTION sys.is_member(IN role sys.SYSNAME)
 RETURNS INT AS
 $$
-	SELECT sys.is_rolemember_internal(role, NULL);
+DECLARE
+    is_windows_auth integer = 0;
+    is_windows_grp boolean := (CHARINDEX('\', role) != 0);
+BEGIN
+    -- Always return 1 for 'public'
+    IF (role = 'public')
+    THEN RETURN 1;
+    END IF;
+
+    SELECT COUNT(*) FROM pg_stat_gssapi WHERE pid = pg_backend_pid() AND gss_authenticated = true INTO is_windows_auth;
+
+    IF EXISTS (SELECT orig_loginname FROM sys.babelfish_authid_login_ext WHERE orig_loginname = role AND type != 'S') -- do not consider sql logins
+    THEN
+        IF EXISTS (SELECT name FROM sys.login_token WHERE name = role AND type NOT IN ('SERVER ROLE', 'SQL LOGIN')) -- do not consider sql logins, server roles
+        THEN RETURN 1; -- Return 1 if current session user is a member of role or windows group
+        ELSIF (is_windows_auth = 0) THEN -- session is not a windows auth session
+            IF (is_windows_grp)
+            THEN RETURN NULL;  -- argument is a windows group
+            ELSE RETURN sys.is_rolemember_internal(role, NULL); -- argument is not a windows group
```

Comment:
1. Explicitly calling is_rolemember_internal is not required IMO. It should be covered by login_token and user_token view.
2. Here we are changing the logic for is_member(), Let's evaluate the performance impact to avoid rework OR Just enable this new logic for windows authenticated session.
3. Let's add a case for long role/user names(more than 64 char length).